pkg/metadata/http.go

```diff
@@ -0,0 +1,131 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"github.com/apache/camel-k/pkg/apis/camel/v1alpha1"
+	"regexp"
+	"strings"
+)
+
+var httpURIs = map[string]bool{
+	"ahc":                  true,
+	"ahc-ws":               true,
+	"atmosphere-websocket": true,
+	"cxf":         true,
+	"cxfrs":       true,
+	"grpc":        true,
+	"jetty":       true,
+	"netty-http":  true,
+	"netty4-http": true,
+	"rest":        true,
+	"restlet":     true,
+	"servlet":     true,
+	"spark-rest":  true,
+	"spring-ws":   true,
+	"undertow":    true,
+	"websocket":   true,
+	"knative":     true,
+}
+
+var passiveURIs = map[string]bool{
+	"bean":       true,
+	"binding":    true,
+	"browse":     true,
+	"class":      true,
+	"controlbus": true,
+	"dataformat": true,
+	"dataset":    true,
+	"direct":     true,
+	"direct-vm":  true,
+	"language":   true,
+	"log":        true,
+	"mock":       true,
+	"properties": true,
+	"ref":        true,
+	"seda":       true,
+	"stub":       true,
+	"test":       true,
+	"validator":  true,
+	"vm":         true,
```

Comment:
I think such information can be included in the camel catalog and endpoint metadata, do you mind raising an issue at camel side ?

Reply:
maybe we can also add some "hacks" to our catalog generator so we do not need to hard code bits in operator' source

Reply:
Done: https://issues.apache.org/jira/projects/CAMEL/issues/CAMEL-12981

Reply:
And https://github.com/apache/camel-k/issues/264